src/main/java/calculator/domain/Parser.java

```diff
@@ -0,0 +1,72 @@
+package calculator.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Parser {
+
+    private static final String NEW_LINE = "\n";
+    private static final String DOUBLE_SLASH = "//";
+    private static final String START_REGEX = "[";
+    private static final String END_REGEX = "]";
+    private static final int LIMIT_MINIMUM_NUMBER = 0;
+
+    private final String operation;
+    private final List<String> delimiters;
+
+    public Parser(String text, List<String> delimiters) {
+        this.operation = trimDelimiter(text);
+        this.delimiters = new ArrayList<>(delimiters);
+    }
+
+    private String trimDelimiter(String text) {
+        if (isCustomDelimiter(text)) {
+            return text.split(NEW_LINE)[1];
+        }
+        return text;
+    }
+
+    private boolean isCustomDelimiter(String text) {
+        return text.startsWith(DOUBLE_SLASH);
+    }
+
+    public List<Integer> splitNumbersByDelimiter() {
+        final String regex = makeRegularExpression();
+        endByDelimiter(regex);
+        final String[] numbers = operation.split(regex);
+        return castToInteger(numbers);
+    }
+
+    private void endByDelimiter(String regex) {
+        if (operation.substring(operation.length() - 1).matches(regex)) {
+            throw new RuntimeException("[ERROR] 숫자 이외의 값을 계산할 수 없습니다.");
+        }
+    }
+
+    private List<Integer> castToInteger(String[] numbers) {
+        try {
+            return Arrays.stream(numbers)
+                .map(value -> {
+                    final Integer number = Integer.parseInt(value);
+                    validPositiveNumber(number);
+                    return number;
+                }).collect(Collectors.toList());
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("[ERROR] 숫자 이외의 값을 계산할 수 없습니다.");
+        }
+    }
+
+    private void validPositiveNumber(Integer value) {
+        if (value < LIMIT_MINIMUM_NUMBER) {
+            throw new IllegalArgumentException("[ERROR] 음수 값은 입력할 수 없습니다.");
+        }
+    }
+
+    private String makeRegularExpression() {
+        return delimiters.stream()
+            .reduce(START_REGEX, (cum, value) -> cum + value)
+            + END_REGEX;
```

Comment:
해당 함수가 바로 이해하기가 힘든 것 같아요. 🤔 
적절히 변수로 추출하여서 가독성을 높여보면 어떨까요? 😃

아래와 같이 변경해볼 수 있을 것 같아요! 😉
```suggestion
        String regex = String.join("", delimiters);
        return START_REGEX + regex + END_REGEX;
```

Reply:
reduce 함수 내 람다식 변수로 `cum`, `value`를 사용하고 계신데 `cum`은 어떤 의미인가요? 🤔

Reply:
제안해주신 방법으로 변경하였습니다.